docs/USAGE-GUIDE.md

```diff
@@ -1777,6 +1777,216 @@ handle. The handle is considered invalid after **s2n_connection_free** is used.
+
+**s2n_config_set_psk_mode** sets the PSK mode on the config object. The PSK mode represents how the pre-shared key was generated and shared between the server and the client. The enum **s2n_psk_mode** lists the supported PSK modes. Use this API prior to the ClientHello to set the PSK mode for the connection. This API can also be used to switch between the different PSK modes.
+
+**s2n_connection_set_psk_mode** sets the PSK mode on the connection object. The supported PSK modes are listed in the
+enum **s2n_psk_mode** within the **api/s2n.h** file. This API overrides the PSK mode set on config for this connection.
+If the connection already contains a list of PSKs **s2n_connection_set_psk_mode** will verify that the PSK mode being set matches
+that of the list. If the PSK mode being set differs from the existing list, an error is returned. If there is no existing list of PSKs present on the connection, the PSK mode of the list is set to the input PSK mode.
+
+### Obtaining Negotiated Pre-Shared Keys
+
+The following APIs enable the caller to obtain the PSK that was selected by the server for the connection:
+
+```c
+int s2n_connection_get_negotiated_psk_identity_length(struct s2n_connection *conn, uint16_t *identity_length);
+int s2n_connection_get_negotiated_psk_identity(struct s2n_connection *conn, uint8_t *identity, uint16_t max_identity_length);
+```
+
+**s2n_connection_get_negotiated_psk_identity_length** gets the negotiated PSK identity length from the connection object.
+If the negotiated PSK does not exist, the value **0** is returned. **s2n_connection_get_negotiated_psk_identity_length** can
+be used to determine if a negotiated PSK exists or not. If the negotiated PSK  exists **s2n_connection_get_negotiated_psk_identity_length** returns a value greater than zero.
+
+**s2n_connection_get_negotiated_psk_identity** gets the negotiated PSK identity from the connection object.
+If the negotiated PSK does not exist, the PSK identity will not be obtained and no error will be returned.
+
+### Selecting a Pre-Shared Key
+
+The following APIs are use to create and free the offered PSK list:
+
+```c
+struct s2n_offered_psk* s2n_offered_psk_new();
+int s2n_offered_psk_free(struct s2n_offered_psk **psk);
+```
+
+**s2n_offered_psk** creates a new offered PSK object. An offered PSK represents a single PSK offered by the client.
+This object is used to create the list of offered PSKs sent by the client. Use **s2n_offered_psk_free** to free
```

Comment:
This object is used to create the list of offered PSKs sent by the client. -> Use this object to create the list of offered PSKs sent by the client.